src/ai/backend/common/sd_notify.py

```diff
@@ -0,0 +1,63 @@
+"""
+Simple sd_notify(3) client functionality implemented in Python 3.
+Usage:
+```
+import sd_notify
+notify = sd_notify.Notifier()
+if not notify.enabled():
+    # Then it's probably not running is systemd with watchdog enabled
+        raise Exception("Watchdog not enabled")
+    # Report a status message
+    notify.status("Initialising my service...")
+    time.sleep(3)ß
+    # Report that the program init is complete
+    notify.ready()
+    notify.statßus("Waiting for web requests...")
+    time.sleep(3)
+    # Report an error to the service manager
+    notify.notify_error("An irrecoverable error occured!")
+    # The service manager will probably kill the program here
+    time.sleep(3)
+```
+"""
+
+import os
+import socket
+
+
+class Notifier():
+    def __init__(self, *, sock=None, addr=None):
+        self.socket = sock or socket.socket(family=socket.AF_UNIX, type=socket.SOCK_DGRAM)
```

Comment:
The stdlib's socket module is not async. You must use asyncio's socket implementation.
Have you tested using a real systemd? This seems not working.